lib/rake/firebase.rake

```diff
@@ -3,6 +3,14 @@ require 'cdo/hip_chat'
 require 'cdo/rake_utils'
 
 namespace :firebase do
+  desc 'Compile firebase security rules and store them in the apps package.'
+  task :compile_rules do
+    # This task can't be run from production, which doesn't have npm installed.
+    # Instead, production should download an apps package containing the compiled rules.
+    raise 'cannot compile firebase security rules without npm' if RakeUtils.system_('which npm') != 0
+    Dir.chdir(apps_dir) { RakeUtils.system 'grunt compile-firebase-rules' }
```

Comment:
You call out that npm is needed, but you don't depend on an `npm install` anywhere in this task.  Should we do that so that this will work even if it's the first thing you run on a new instance?